tests/xcodeproj/BUILD.bazel

```diff
@@ -14,13 +15,56 @@ ios_unit_test(
     sdk_frameworks = ["XCTest"],
 )
 
+default_ios_test_runner(
+    name = "iPhone-X__13.2",
+    device_type = "iPhone X",
+    os_version = "13.2",
+    visibility = ["//visibility:public"],
+)
+
+default_ios_test_runner(
+    name = "iPad-Air-2__12.4",
+    device_type = "iPad Air 2",
+    os_version = "12.4",
+    visibility = ["//visibility:public"],
+)
+
+ios_unit_test(
+    name = "Single-Application-RunnableTestSuite",
+    srcs = ["test.swift"],
+    args = ["-h"],
+    env = {
+        "test_envvar_key1": "test_envvar_value1",
+        "test_envvar_key2": "test_envvar_value2",
+    },
+    minimum_os_version = "11.0",
+    module_name = "Single_Application_RunnableTestSuite",
+    runners = [
+        "iPhone-X__13.2",
+        "iPad-Air-2__12.4",
+    ],
+    sdk_frameworks = ["XCTest"],
+)
+
+ios_unit_test(
+    name = "Single-Application-UnrunnableTestSuite",
+    srcs = ["test.swift"],
+    minimum_os_version = "11.0",
+    module_name = "Single_Application_UnrunnableTestSuite",
+    runners = [],
+    sdk_frameworks = ["XCTest"],
+)
+
 xcodeproj(
     name = "Single-Application-Project",
     testonly = True,
     bazel_path = "bazelisk",
+    include_transitive_targets = True,
```

Comment:
Is this a new param available from xcodeproj rule? Or it is from somewhere else? I am always confused about this because after all it (to me) just shows up in the middle of no where :)

Reply:
No, this param/attribute was already available in the xcodeproj rule. It has a default value of false.

This bug is only exhibited when this attribute is set to True.

If `include_transitive_targets` is false, the xcodeproject rule reads the "target" property of the provider returned by the aspect. This target just refers to a single target.

But if `include_transitive_targets` is true, the xcodeproject rule reads the "all_deps_target" property of the provider returned by the aspect. (I renamed "targets" to "all_deps_target", which is a more descriptive name imo.) The "all_deps_target" property refers to all the targets that this target depends on (directly and transitively.)

This PR fixes a bug. When the aspect is applied to a test_suite, prior to my changes, the "target" property of the provider would refer to the test_suite, but the "all_deps_target" did not refer to the test_suite itself ... and as a result, the xcodeproject would also fail to pick up the test_suite.

Reply:
Should we make a new project that sets `include_transitive_targets = True`, so we can have test coverage of both cases?

Reply:
Thx for the explaination!